src/main/java/carpet/script/value/FormattedTextValue.java

```diff
@@ -14,6 +15,33 @@ public FormattedTextValue(Text text)
         this.text = text;
     }
 
+    public static Value combine(Value left, Value right) {
+        BaseText text;
+        if (left instanceof FormattedTextValue)
+        {
+            text = (BaseText) ((FormattedTextValue) left).getText().shallowCopy();
+        }
+        else
+        {
+            if (left instanceof NullValue)
+                return right;
```

Comment:
Should I use `getString() == null` instead?

As far as I understood, `getString()` can't ever be null, since `NullValue` would return `"null"` as the String, that's why I changed it.

Also should I clone the variables before returning them in those cases?

Reply:
left.isNull() should be preferred.

Reply:
but it doesn't matter that much. I wanted to stop using instanceof, since isNull may get more complex in the future. Don't worry about it.